src/org/opensolaris/opengrok/index/IndexAnalysisSettingsAccessor.java

```diff
@@ -70,20 +86,24 @@ public IndexAnalysisSettings read(IndexReader reader) throws IOException {
             // This is not expected, so translate to RuntimeException.
             throw new RuntimeException(ex);
         }
-        TopDocs top = searcher.search(q, 1);
-        if (top.totalHits < 1) {
-            return null;
-        }
+        TopDocs top = searcher.search(q, n);
 
-        Document doc = searcher.doc(top.scoreDocs[0].doc);
-        IndexableField objser = doc.getField(QueryBuilder.OBJSER);
-        try {
-            return objser == null ? null : IndexAnalysisSettings.deserialize(
-                objser.binaryValue().bytes);
-        } catch (ClassNotFoundException ex) {
-            // This is not expected, so translate to RuntimeException.
-            throw new RuntimeException(ex);
+        int nres = top.totalHits > n ? n : (int)top.totalHits;
```

Comment:
thank you for this update, it will make our life easier with new releases of lucene, where they prefer this new approach to fetching results